app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -349,6 +352,24 @@ public void init() {
         fieldFlowable.connect();
     }
 
+    @VisibleForTesting
+    public String getNextVisibleSection(String activeSection, List<FormSectionViewModel> sectionList) {
+        while (sectionsToHide.contains(activeSection)) {
+            for (FormSectionViewModel section : sectionList) {
+                if (section.sectionUid().equals(activeSection)) {
+                    int nextSectionIndex = sectionList.indexOf(section) + 1;
+                    if (nextSectionIndex < sectionList.size()) {
+                        activeSection = sectionList.get(nextSectionIndex).sectionUid();
+                    }else{
+                        activeSection = "";
+                    }
+                    break;
+                }
+            }
+        }
+        return activeSection;
+    }
```

Comment:
I think this could go to another class and maybe reuse in the future? The fact that you need to add @VisibleForTesting here tells you too many things are going on in the flatmap. Pls do not make an extension :+1:

Reply:
Also it could be done in Kotlin with collection operations